### PR TITLE
[FEAT] History Maintenance

### LINKED ISSUE #76: Pencatatan Riwayat Maintenance
# Overview
Untuk meningkatkan manajemen alat medis dan pemeliharaan, diperlukan sebuah fitur yang dapat mencatat dan menampilkan riwayat semua aktivitas maintenance yang telah dilakukan pada setiap alat medis. Ini akan membantu dalam melacak kondisi alat medis, teknisi pemelihara alat medis tersebut, dan tanggal terakhir dilakukan pemeliharaan.

## Intended Outcome
Setelah fitur ini diimplementasikan, pengguna dapat dengan mudah melihat histori lengkap dari semua tindakan pemeliharaan yang pernah dilakukan pada suatu alat medis tertentu. Informasi ini dapat diakses melalui halaman detail alat medis.

## Tasks
- [x] Menambahkan modul agar bisa POST dan GET di sistem untuk menyimpan dan menampilkan data riwayat maintenance (tanggal, teknisi, catatan).
- [x] Integrasi UI di Frontend untuk menampilkan daftar riwayat maintenance pada halaman detail alat medis. Daftar harus urut berdasarkan tanggal maintenance (DESC).
- [x] Memastikan data riwayat maintenance tersimpan dengan benar di database dan terhubung dengan alat medis yang sesuai.
- [x] Pembuatan Unit Tests untuk memvalidasi fungsi pencatatan dan pengambilan data riwayat.

## Acceptance Criteria
- Pengguna dapat membuka halaman detail aset dan melihat tab/bagian "Riwayat Maintenance".
- Daftar riwayat maintenance menampilkan minimal: Tanggal, Nama Teknisi, dan Catatan untuk Maintenance yang dilakukan.
- Data yang ditampilkan pada riwayat maintenance sesuai dengan data yang dimasukkan/tersimpan di database.
- Semua unit test yang terkait dengan fitur ini berhasil dijalankan (status: Passed/GREEN).
- Fitur berfungsi tanpa menyebabkan error pada bagian lain sistem.

### LINKED ISSUE #92: Create Unit Tests for Maintenance History
Unit Test for Maintenance History with TDD Approach

### LINKED ISSUE #93: Add Implementations for Maintenance History
Add Implementations for Maintenance History Feature and should already be tested with Postman

### PR DESCRIPTION
**Pull request by daffagrahito**
# Summary

Closes #92 : This task is closed based on commit hash `9725624`.
Closes #93 : This task is closed based on commit hash  `f36d9176` dan `28497282`.

This pull request introduces extensive unit tests for maintenance history and medical equipment functionality, along with consistent naming updates for medical equipment-related files. The key changes include new implementations for Maintenance History feature.

### New Unit Tests for Maintenance History:

* [`__tests__/unit/controllers/maintenance-history/create-maintenance-history.controller.unit.test.ts`](diffhunk://#diff-77554ca413f0379fe9c999e95484db114d167af7417cf8d935a09b932b09c686R1-R134): Added tests for the `createMaintenanceHistory` method in `MaintenanceHistoryController`, covering positive cases, error handling, and edge cases like missing fields and undefined users.
* [`__tests__/unit/controllers/maintenance-history/get-maintenance-history.controller.unit.test.ts`](diffhunk://#diff-5d18b7ddb8ff3128a82f663b5a2202d7ef2d08582834464b665add59d2b1bedcR1-R393): Added tests for the `getMaintenanceHistoriesByEquipmentId` method in `MaintenanceHistoryController`, including scenarios with pagination, filters, search parameters, and error handling.
* [`__tests__/unit/repository/maintenance-history/create-maintenance-history.repository.unit.test.ts`](diffhunk://#diff-d863f2ec5cd272eabc78079c143c060f10ee02f0cd13f618695b338a6c143ae8R1-R167): Added tests for the `createMaintenanceHistory` method in `MaintenanceHistoryRepository`, validating database interactions, error handling, and date conversion logic.

### File Renaming for Medical Equipment:

* Renamed medical equipment-related test files and updated imports to use `medical-equipment` instead of `medicalequipment`, ensuring consistency across the codebase. This change affects files for adding, deleting, updating, and retrieving medical equipment. [[1]](diffhunk://#diff-7c8e3c98c9433018ae554628db87d28bacd16458f6551627a9f3d83cd2d12b02L1-R5) [[2]](diffhunk://#diff-66f317454ae604ca7359ef06f5667a93fd9861340f6c2388a17bffdd9d4e9cbbL1-R5) [[3]](diffhunk://#diff-96f342ef6c583c0cb55115c9a44c48884e95c9b41a8c2a68814a68bbaea91c3eL1-R5) [[4]](diffhunk://#diff-b26ff52f78258bcb917e24cfa442cc52ff241010b764e303200024666031aa3fL1-R5) [[5]](diffhunk://#diff-5d077437cea297a075aab16a1e75fdfab642db017cbfef3df934b7dada850952L1-R1) [[6]](diffhunk://#diff-dc31956b65006fdfc0f26aae59d62ca367905a964d0ec21251ca76dd1a46601fL1-R1)

## Next Step
* Add implementations and unit tests for Calibration History
* Add implementations and unit tests for Parts History (Management Parts)

## Summary by CodeRabbit

- **New Features**
  - Added maintenance history tracking for medical equipment, including the ability to create and view maintenance records with filtering and pagination.
  - Introduced new API endpoints for managing and retrieving maintenance history data.
  - Implemented validation for maintenance history creation and filtering requests.

- **Database**
  - Added new tables for maintenance, calibration, and parts history; removed legacy parts management tables.
  - Updated table structures and relationships to support detailed tracking of maintenance and calibration events.

- **Documentation & Tests**
  - Added comprehensive unit tests for maintenance history controllers, services, repositories, and validation logic.
  - Updated import paths for consistency and maintainability.

### CODE REVIEW

**Comment by daffagrahito** [comment]
@coderabbitai review

File: src/validations/maintenance-history.validation.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    .withMessage("Result is required")
+    .isIn(["Success", "Partial", "Failed"])
+    .withMessage("Result must be Success, Partial, or Failed"),
+  body("maintenanceDate")
+    .notEmpty()
+    .withMessage("Maintenance date is required")
+    .isISO8601()
+    .withMessage("Invalid date format")
+    .customSanitizer((value) => toJakartaDate(value)),
+];
```
[the remark this answers is not in the document]
Comment (daffagrahito) [reply src/validations/maintenance-history.validation.ts:23]: not needed, because it isn't defined that way in the schema
